lib/rubocop/cop/rspec/expect_in_let.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Do not use `expect` in let.
+      #
+      # @example
+      #   # bad
+      #   let(:foo) do
+      #     expect(something).to eq 'foo'
+      #   end
+      #
+      #   # good
+      #   it do
+      #     expect(something).to eq 'foo'
+      #   end
+      #
+      class ExpectInLet < Base
+        MSG = 'Do not use `%<expect>s` in let'
+
+        # @!method expectation(node)
+        def_node_search :expectation, '(send nil? #Expectations.all ...)'
+
+        def on_block(node)
+          return unless let?(node)
+          return if node.body.nil?
+
+          expectation(node.body) do |expect|
+            add_offense(expect.loc.selector, message: message(expect))
+          end
```

Comment:
`#let?` is a node matcher, so you can use the block form instead of a guard clause:
```suggestion
          return if node.body.nil?
          let?(node) do
            expectation(node.body) do |expect|
              add_offense(expect.loc.selector, message: message(expect))
            end
          end
```

Reply:
I think it’s how it is done in ExpectInHook.
I’d keep for conformity.

Reply:
Ah yes, I didn’t notice. Let’s keep it this way.